posthog/models/team/team.py

```diff
@@ -345,6 +348,9 @@ def person_on_events_mode_flag_based_default(self) -> PersonsOnEventsMode:
         if self._person_on_events_person_id_no_override_properties_on_events:
             return PersonsOnEventsMode.PERSON_ID_NO_OVERRIDE_PROPERTIES_ON_EVENTS
 
+        if self.organization.created_at >= "2024-06-14":
+            return PersonsOnEventsMode.PERSON_ID_OVERRIDE_PROPERTIES_ON_EVENTS
```

Comment:
This should already be handled by the [`persons-on-events-v2-reads-enabled`](https://us.posthog.com/project/2/feature_flags/8523) flag – is that not the case? As in, this `person_on_events_mode_flag_based_default` property should be short-circuited at `_person_on_events_person_id_override_properties_on_events`, which should be true for teams in orgs created on or after 2024-06-14